Adding info about docker vs podman extensions (#41837)
quarkusio/quarkus: docs/src/main/asciidoc/container-image.adoc

:add-extension-extensions: container-image-podman
include::{includes}/devtools/extension-add.adoc[]

[TIP]
.When to use Docker vs Podman extension
====
The <<#docker,Docker extension>> is and has always been backwards-compatible with Podman because Podman exposes a https://podman.io/docs/installation[Docker-compatible API]. You can build container images with Podman using the Docker extension (see the https://quarkus.io/guides/podman[Using Podman with Quarkus guide]).

> You need to use either the Docker or Podman specific extension when you need to do things that are specific to either Docker or Podman.

Reviewer: ```suggestion
You need to use either the `quarkus-container-image-docker` or `quarkus-container-image-podman` extension when you need to do things specific to either Docker or Podman.
```

Reviewer: I think specifying the extension names is clearer than stating "Docker or Podman specific extension"

Author: Agreed. I committed your suggestion.

Reviewer: That was fast, I added some minor wording improvements to that sentence as well, see if that makes sense and please squash the commits before merging 😉

Author: I think what you wrote is fine. Can't the commits be squashed when merging? Squash & commit vs Merge & commit?

Reviewer: ![image](https://github.com/user-attachments/assets/e25ede49-200c-465c-ae37-b1ed73e8aa72)

Author: I was going to ask "Why not?", but I'll save that for another day when I have time for a longer debate :) I'll go ahead and rebase now...

Author: Rebase done

Author: Weird when I clicked on "Commit suggestion" it didn't commit the whole suggestion. Just fixed & rebased

Reviewer: It's because of how we backport PRs, squashing and merging would make the commit untraceable
